protocol/messenger_filter_init.go

```diff
@@ -209,32 +205,6 @@ func (m *Messenger) processSingleChat(chat *Chat, communityInfo map[string]*comm
 	return filters, publicKeys, nil
 }
 
-func (m *Messenger) processCommunityChat(chat *Chat, communityInfo map[string]*communities.Community) (transport.FiltersToInitialize, error) {
-	community, ok := communityInfo[chat.CommunityID]
-	if !ok {
-		var err error
-		community, err = m.communitiesManager.GetByIDString(chat.CommunityID)
-		if err != nil {
-			return transport.FiltersToInitialize{}, err
-		}
-		communityInfo[chat.CommunityID] = community
-	}
-
-	if chat.UnviewedMessagesCount > 0 || chat.UnviewedMentionsCount > 0 {
-		// Make sure the unread count is 0 for the channels the user cannot view
-		// It's possible that the users received messages to a channel before permissions were added
-		if !community.CanView(&m.identity.PublicKey, chat.CommunityChatID()) {
-			chat.UnviewedMessagesCount = 0
-			chat.UnviewedMentionsCount = 0
-		}
-	}
```

Comment:
This part is still needed. I would go with following approach:
```diff
diff --git a/protocol/messenger_filter_init.go b/protocol/messenger_filter_init.go
index e66030a00..e31fa690b 100644
--- a/protocol/messenger_filter_init.go
+++ b/protocol/messenger_filter_init.go
@@ -182,11 +182,12 @@ func (m *Messenger) processSingleChat(chat *Chat, communityInfo map[string]*comm
 		filters = append(filters, transport.FiltersToInitialize{ChatID: chat.ID})
 
 	case ChatTypeCommunityChat:
-		filter, err := m.processCommunityChat(chat, communityInfo)
+		// Since universalChatID is being used, no specific filters needs to be registered for all community chats.
+		// Reasoning: https://github.com/status-im/status-go/pull/5993
+		err := m.processCommunityChat(chat, communityInfo)
 		if err != nil {
 			return nil, nil, err
 		}
-		filters = append(filters, filter)
 
 	case ChatTypeOneToOne:
 		pk, err := chat.PublicKey()
@@ -209,13 +210,13 @@ func (m *Messenger) processSingleChat(chat *Chat, communityInfo map[string]*comm
 	return filters, publicKeys, nil
 }
 
-func (m *Messenger) processCommunityChat(chat *Chat, communityInfo map[string]*communities.Community) (transport.FiltersToInitialize, error) {
+func (m *Messenger) processCommunityChat(chat *Chat, communityInfo map[string]*communities.Community) error {
 	community, ok := communityInfo[chat.CommunityID]
 	if !ok {
 		var err error
 		community, err = m.communitiesManager.GetByIDString(chat.CommunityID)
 		if err != nil {
-			return transport.FiltersToInitialize{}, err
+			return err
 		}
 		communityInfo[chat.CommunityID] = community
 	}
@@ -229,10 +230,7 @@ func (m *Messenger) processCommunityChat(chat *Chat, communityInfo map[string]*c
 		}
 	}
 
-	return transport.FiltersToInitialize{
-		ChatID:      chat.ID,
-		PubsubTopic: community.PubsubTopic(),
-	}, nil
+	return nil
 }
```